### PR TITLE
[torch] Add folders for `torch.fill`, `torch.ones`, `torch.zeros` and `aten.getItem`

### PR DESCRIPTION
**Pull request by renxida**
Thanks for the idea @newling

### CODE REVIEW

**Comment by newling** [comment]
Tests in Dialect/Torch/canonicalize.mlir if possible please

I think it's fine but I'm not experienced enough with torch-mlir to sign off on this -- I'm not sure if DenseElementsAttr are supported (I can see folders returning IntegerAttr, none return DenseElementsAttr). Please get another review.

File: include/torch-mlir/Dialect/Torch/IR/GeneratedTorchOps.td [hunk 1]
```diff
@@ -11202,6 +11205,7 @@ def Torch_AtenFullOp : Torch_Op<"aten.full", [
       printDefaultTorchOp(printer, *this, 6, 1);
     }
   }];
+  let hasFolder = 1;
```
Comment (rsuderman) [inline include/torch-mlir/Dialect/Torch/IR/GeneratedTorchOps.td:11208]: You may need to rerun the generator as this does not have an implementation and is no longer specified in the script.

File: lib/Dialect/Torch/IR/TorchOps.cpp [hunk 2]
```diff
@@ -2813,6 +2814,116 @@ OpFoldResult AtenDivIntOp::fold(FoldAdaptor adaptor) {
...
+  if (!shapedty) {
+    LLVM_DEBUG(llvm::dbgs()
+               << "Failing to fold AtenOnesOp: ShapedType cast failed.\n");
+    return nullptr;
+  }
+  auto elementType = shapedty.getElementType();
+  if (elementType.isa<IntegerType>()) {
+    Attribute attribute = IntegerAttr::get(elementType, 1);
+    return DenseElementsAttr::get(shapedty, attribute);
+  } else if (elementType.isa<FloatType>()) {
```
Comment (rsuderman) [inline lib/Dialect/Torch/IR/TorchOps.cpp:2858]: You might as well drop the `else` of each of these statements as each returns a value.

Repeat for the other sections below.

File: test/Dialect/Torch/canonicalize.mlir [hunk 3]
```diff
@@ -29,6 +29,61 @@ func.func @torch.runtime.assert() {
...
+    %1 = torch.aten.zeros %0, %int3, %none, %none, %none : !torch.list<int>, !torch.int, !torch.none, !torch.none, !torch.none -> !torch.vtensor<[1],si32>
+    %2 = torch.aten.item %1 : !torch.vtensor<[1],si32> -> !torch.int
+    return %2 : !torch.int
+}
+
+// test that we can fold the constant when we chain
+// aten.full and aten.item
+// this should return a single number
+// CHECK-LABEL:   func.func @torch.aten.full_item
+// ensure a constant int is created and returned
```
Comment (rsuderman) [inline test/Dialect/Torch/canonicalize.mlir:70]: Don't both including this comment. The check statements are fairly straightforward

File: test/Dialect/Torch/canonicalize.mlir [hunk 4]
```diff
@@ -29,6 +29,61 @@ func.func @torch.runtime.assert() {
...
+    %int1 = torch.constant.int 1
+    %int3 = torch.constant.int 3
+    %none = torch.constant.none
+    %0 = torch.prim.ListConstruct %int1 : (!torch.int) -> !torch.list<int>
+    %1 = torch.aten.zeros %0, %int3, %none, %none, %none : !torch.list<int>, !torch.int, !torch.none, !torch.none, !torch.none -> !torch.vtensor<[1],si32>
+    %2 = torch.aten.item %1 : !torch.vtensor<[1],si32> -> !torch.int
+    return %2 : !torch.int
+}
+
+// test that we can fold the constant when we chain
```
Comment (rsuderman) [inline test/Dialect/Torch/canonicalize.mlir:66]: Same with this comment. The IR is pretty sufficient.

File: test/Dialect/Torch/canonicalize.mlir [hunk 5]
```diff
@@ -29,6 +29,61 @@ func.func @torch.runtime.assert() {
...
+    %int3 = torch.constant.int 1337
+    %int5 = torch.constant.int 5
+    %none = torch.constant.none
+    %0 = torch.prim.ListConstruct %int1 : (!torch.int) -> !torch.list<int>
+    %1 = torch.aten.full %0, %int3, %int5, %none, %none, %none : !torch.list<int>, !torch.int, !torch.int, !torch.none, !torch.none, !torch.none -> !torch.vtensor<[1],si32>
+    %2 = torch.aten.item %1 : !torch.vtensor<[1],si32> -> !torch.int
+    return %2 : !torch.int
+}
+
+// test that we can fold the constant when we chain
```
Comment (rsuderman) [inline test/Dialect/Torch/canonicalize.mlir:84]: Ditto for removing.

**Review by rsuderman — CHANGES_REQUESTED** [review]

**Review by renxida — COMMENTED** [review]
addressed!

**Review by rsuderman — APPROVED** [review]

**Pull request merged**